client/components/account-balances/index.tsx

```diff
@@ -1,28 +1,265 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { Card } from '@wordpress/components';
+import * as React from 'react';
+import { Flex, TabPanel } from '@wordpress/components';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import interpolateComponents from '@automattic/interpolate-components';
 
 /**
  * Internal dependencies
  */
-import AccountBalancesHeader from './header';
+import { useAllDepositsOverviews } from 'wcpay/data';
+import { useSelectedCurrency } from 'wcpay/overview/hooks';
+import { getCurrencyTabTitle } from './utils';
+import BalanceBlock from './balance-block';
+import BalanceTooltip from './balance-tooltip';
+import { documentationUrls, fundLabelStrings } from './strings';
+import InstantDepositButton from 'deposits/instant-deposits';
+import wcpayTracks from 'tracks';
 import './style.scss';
-import AccountBalancesTabPanel from './balances-tab-panel';
 
 /**
- * Renders an overview of the account's balances.
+ * BalanceTab
  *
- * @return {JSX.Element} Rendered element with the account balances card.
+ * @typedef {Object} BalanceTab
+ *
+ * @param {string} name           Name of the tab.
+ * @param {string} title          Title of the tab.
+ * @param {string} currencyCode   Currency code of the tab.
+ * @param {number} availableFunds Available funds of the tab.
+ * @param {number} pendingFunds   Pending funds of the tab.
+ * @param {number} delayDays	  The account's pending period in days.
+ */
+type BalanceTab = {
+	name: string;
+	title: string;
+	currencyCode: string;
+	availableFunds: number;
+	pendingFunds: number;
+	delayDays: number;
+	instantBalance?: AccountOverview.InstantBalance;
+};
+
+/**
+ * Renders an account balances panel with tab navigation for each deposit currency.
+ *
+ * @return {JSX.Element} Rendered balances panel with tab navigation for each currency.
  */
 const AccountBalances: React.FC = () => {
-	return (
-		<Card className="wcpay-account-balances">
-			<AccountBalancesHeader />
+	const {
+		overviews,
+		isLoading,
+	} = useAllDepositsOverviews() as AccountOverview.OverviewsResponse;
```

Comment:
What does the `as AccountOverview.OverviewsResponse` do? It looks like a cast, but `useAllDepositsOverviews()` returns that type anyway.

Reply:
This is true, removed in 0f33c1a92a370d2fe01e497fb09515e905318a66.

Reply:
This is true, removed in 0f33c1a92a370d2fe01e497fb09515e905318a66.